src/classes/TDTM_ObjectDataGateway.cls

```diff
@@ -54,45 +54,51 @@ public class TDTM_ObjectDataGateway implements TDTM_iTableDataGateway {
             return true;    
         return false;
     }
-    
+
     /*******************************************************************************************************
      * @description Returns the classes that will be called for the specified ojbect and action. Returned list 
-     *              is sorted by load order.
+     *              is sorted by load order. Filters out in-active handlers and any where the list of 
+     *              excluded usernames includes the current context user.
      * @param objectName The name of the object.
      * @param action The action (before insert, after insert,...). 
      * @return List<SObject> List of classes to call, sorted by load order.
      **/
     public static List<SObject> getClassesToCallForObject(String objectName, TDTM_Runnable.Action action) {   
         string strAction = action.Name();
+        string currUserName = UserInfo.getUserName();
```

Comment:
True. I generally think of any calls to system classes as expensive, however this is a very small recordset, so it be negligible.